metricbeat/helper/prometheus/module.go

```diff
@@ -56,6 +56,7 @@ type prometheusMetricSet struct {
 	mapping    *MetricsMapping
 }
 
-func (m *prometheusMetricSet) Fetch(r mb.ReporterV2) {
+func (m *prometheusMetricSet) Fetch(r mb.ReporterV2) error {
 	m.prometheus.ReportProcessedMetrics(m.mapping, r)
+	return nil
```

Comment:
Yah, it's awkward, but I ran into some issues if the helper module wasn't the error interface type. I was hoping to convert _everything_ since that way we could remove the old/redundant code if we wanted to.

Reply:
I guess that in theory we could keep old interface for metricsets that cannot return errors. But does such a thing exist?

We could check if `m.prometheus.ReportProcessedMetrics()` is hiding some error and return it to handle or return it here.

Reply:
> We could check if m.prometheus.ReportProcessedMetrics() is hiding some error and return it to handle or return it here.

I was considering that, but I was worried I might break even more things, hah. Let me give it a shot.